### PR TITLE
Feature/sc 1641/upgrade web3 react package to v8

### PR DESCRIPTION
**Pull request by adamgoth**
### Link to Shortcut ticket:
https://app.shortcut.com/dux-makerdao/story/1641/upgrade-web3-react-package-to-v8

### What does this PR do?
Updates web3-react from v6 to v8
Adds gnosis safe as an option

### Steps for testing:
Try the different wallet options
Try switching and disconnecting etc

### CODE REVIEW

**Comment by b-pmcg** [comment]
I noticed that when clicking "disconnect" from the Account box, and then refreshing the page the app will eagerly connect the account again. Would removing the connection from the array in `useOrderedConnections` be enough to prevent that?

File: next.config.js [hunk 1]
```diff
@@ -15,15 +15,32 @@ const securityHeaders = [
...
 
   // adds x-frame-options
-  {
-    key: 'X-Frame-Options',
-    value: 'SAMEORIGIN'
-  },
+  // remove for safe app to load
+  // {
+  //   key: 'X-Frame-Options',
+  //   value: 'SAMEORIGIN'
```
Comment (rafinskipg) [inline next.config.js:21]: Should we restore this?

**Review by rafinskipg — DISMISSED** [review]
waiting on the link to the modified package from safe-react

File: next.config.js (hunk 1 above)
Comment (adamgoth) [reply next.config.js:21]: We won't be able to load the app in the safe app iframe with it set. 

> if you specify SAMEORIGIN, you can still use the page in a frame as long as the site including it in a frame is the same as the one serving the page.

I'm not sure when or why it got added. I guess click-jacking is perhaps the security concern.

https://www.keycdn.com/blog/x-frame-options

**Review by b-pmcg — APPROVED** [review]
LFG!

**Pull request merged**